src/connector/codec/tests/integration_tests/utils.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub use expect_test::{expect, Expect};
+pub use itertools::Itertools;
+pub use risingwave_common::catalog::ColumnDesc;
+use risingwave_common::types::{
+    DataType, Datum, DatumCow, DatumRef, ScalarImpl, ScalarRefImpl, ToDatumRef,
+};
+use risingwave_pb::plan_common::AdditionalColumn;
+
+/// More concise display for `DataType`, to use in tests.
+pub struct DataTypeTestDisplay<'a>(pub &'a DataType);
+
+impl<'a> std::fmt::Debug for DataTypeTestDisplay<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.0 {
+            DataType::Struct(s) => {
+                // since this is for tests, we don't care about perf here :)
+                if !f.alternate() || s.len() == 1 {
+                    let (name, ty) = s.iter().next().unwrap();
+                    return write!(f, "Struct {{ {}: {:?} }}", name, &DataTypeTestDisplay(ty));
```

Comment:
Why do we only display the first field of a struct?

Reply:
😄